contracts/src/rollup/RollupAdminLogic.sol

```diff
@@ -324,4 +324,34 @@ contract RollupAdminLogic is RollupCore, IRollupAdmin, SecondaryLogicUUPSUpgrade
         inbox = newInbox;
         emit OwnerFunctionCalled(28);
     }
+
+    function createNitroMigrationGenesis(uint64 genesisBlockNumber, bytes32 genesisBlockHash)
+        external
+        whenPaused
+    {
+        require(latestNodeCreated() == 0, "NON_GENESIS_NODES_EXIST");
+        bytes32[2] memory newStateBytes32;
+        newStateBytes32[0] = genesisBlockHash;
+        uint64[2] memory newStateU64;
+        newStateU64[0] = 1;
+        GlobalState memory emptyGlobalState;
+        RollupLib.Assertion memory assertion = RollupLib.Assertion({
+            beforeState: RollupLib.ExecutionState({
+                globalState: emptyGlobalState,
+                machineStatus: MachineStatus.FINISHED
+            }),
+            afterState: RollupLib.ExecutionState({
+                globalState: GlobalState({bytes32Vals: newStateBytes32, u64Vals: newStateU64}),
+                machineStatus: MachineStatus.FINISHED
+            }),
+            numBlocks: genesisBlockNumber + 1
+        });
+        // TODO: hacky workaround for the createNewNode assertion argument being in calldata
```

Comment:
we could have the assertion be a param instead (and we do a couple requires to be sure its what we expect

```solidity
    function createNitroMigrationGenesis(Assertion calldata assertion) external whenPaused {
        require(assertion.beforeState.globalState[0] == emptyGlobalState[0], "");
        // ... 
        createNewNode(assertion, 0, 1, bytes32(0));
```